fdk/context.py

```diff
@@ -115,48 +91,25 @@ def Deadline(self):
             return now.isoformat()
         return self.__deadline
 
-    def SetResponseHeaders(self, headers, status_code, content_type=None):
-        self.__response_headers = set_response_headers(
-            self.GetResponseHeaders(), headers, status_code,
-            content_type=content_type)
+    def SetResponseHeaders(self, headers, status_code):
+        if self.isGateway():
+            headers = hs.encap_headers(headers, status=status_code)
+
+        for k, v in headers.items():
+            self.__response_headers[k] = v
 
     def GetResponseHeaders(self):
         return self.__response_headers
 
-    def HTTPContext(self):
-        return HTTPGatewayContext(self)
-
-
-class HTTPGatewayContext(object):
-    def __init__(self, invoke_context: InvokeContext):
-        """
-        Creates an instance of an HTTP context
-        :param invoke_context: invoke context:
-        :type invoke_context: fdk.context.InvokeContext
-        """
-        self.__headers = hs.decap_headers(invoke_context.Headers())
-        self.__invoke_context = invoke_context
-        self.__response_headers = {}
-
     def RequestURL(self):
         return self.__invoke_context._request_url
 
     def Method(self):
         return self.__invoke_context._method
 
-    def Headers(self):
-        return self.__headers
-
-    def SetResponseHeaders(self, headers, status_code, content_type=None):
-        self.__response_headers = set_response_headers(
-            self.GetResponseHeaders(), headers, status_code,
-            content_type=content_type)
-
-    def GetResponseHeaders(self):
-        return self.__response_headers
-
-    def Format(self):
-        return self.__invoke_context.Format()
+    def isGateway(self):
```

Comment:
if this method supposes to be hidden from a user, it must start with a double underscore, i.e.,:
```python
def __is_gateway(self)
```

Reply:
ah, thanks. now that I think about it, it may be useful for users since http gateway context and invoke contexts don't have a different type (and InvokeContext name kind of confusing) - though idk that we want to support it so maybe we should hide it. I think we should probably have separate types for them like it was before?